### PR TITLE
Federation members kicker and bugfixes

### PR DESCRIPTION
**Pull request by noescape00**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.PoA/Voting/FederationVotingController.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        protected readonly IFederationManager fedManager;
+
+        protected readonly VotingManager votingManager;
+
+        protected readonly Network network;
+
+        protected readonly ILogger logger;
+
+        public FederationVotingController(IFederationManager fedManager, ILoggerFactory loggerFactory, VotingManager votingManager, Network network)
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.PoA/Voting/FederationVotingController.cs]: Shouldn't `GetFederationMembers` also go in this controller?

File: src/Stratis.Bitcoin.Features.PoA/Voting/IdleFederationMembersKicker.cs [hunk 2]
```diff
@@ -63,17 +70,26 @@ public void Initialize()
             this.fedMemberAddedToken = this.signals.Subscribe<FedMemberAdded>(this.OnFedMemberAdded);
             this.fedMemberKickedToken = this.signals.Subscribe<FedMemberKicked>(this.OnFedMemberKicked);
 
-            this.fedPubKeysByLastActiveTime = this.keyValueRepository.LoadValueJson<Dictionary<PubKey, uint>>(fedMembersByLastActiveTimeKey);
+            Dictionary<string, uint> loaded = this.keyValueRepository.LoadValueJson<Dictionary<string, uint>>(fedMembersByLastActiveTimeKey);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.PoA/Voting/IdleFederationMembersKicker.cs]: Why string and not pubkey?

Comment (noescape00) [reply src/Stratis.Bitcoin.Features.PoA/Voting/IdleFederationMembersKicker.cs]: Problem with serialization. Using string solves the problem.

File: src/Stratis.Bitcoin.Features.PoA/Voting/FederationVotingController.cs (hunk 1 above)
Comment (noescape00) [reply src/Stratis.Bitcoin.Features.PoA/Voting/FederationVotingController.cs]: `GetFederationMembers` is in base controller. This method is exactly the same for federation and normal PoA so I saw no reason to put it here.

**Pull request merged**